botocore/session.py

```diff
@@ -919,6 +923,11 @@ def create_client(
         :type aws_account_id: string
         :param aws_account_id: The AWS account ID to use when creating the client.
             Same semantics as aws_access_key_id above.
+
+        :type aws_credential_scope: string
+        :param aws_credential_scope: The AWS credential scope to use when creating
+            creating the client. Same semantics as aws_access_key_id above. Must be
+            equal to region_name.
```

Comment:
> Must be equal to region_name.

I think we discussed this not necessarily being true. We should notify through the exceptions rather than hardcoding this in the description.